src/graph/graph_op.cc

```diff
@@ -315,4 +315,74 @@ ImmutableGraph GraphOp::ToSimpleGraph(const GraphInterface* graph) {
   return ImmutableGraph(csr);
 }
 
+Graph GraphOp::ToBidirectedMutableGraph(const GraphInterface* g) {
+  std::unordered_map<int, std::unordered_map<int, int>> n_e;
+  for (dgl_id_t u = 0; u < g->NumVertices(); ++u) {
+    for (const dgl_id_t v : g->SuccVec(u)) {
+      n_e[u][v]++;
+    }
+  }
+
+  Graph bg;
+  bg.AddVertices(g->NumVertices());
+  for (dgl_id_t u = 0; u < g->NumVertices(); ++u) {
+    for (dgl_id_t v = u; v < g->NumVertices(); ++v) {
+      const auto new_n_e = std::max(n_e[u][v], n_e[v][u]);
+      if (new_n_e > 0) {
+        IdArray us = IdArray::Empty({new_n_e}, DLDataType{kDLInt, 64, 1}, DLContext{kDLCPU, 0});
```

Comment:
Try use the new routines in `include/dgl/array.h`.

Reply:
Done.